pkg/build/builder/dockerutil.go

```diff
@@ -89,6 +120,35 @@ func pushImage(client DockerClient, name string, authConfig docker.AuthConfigura
 	return err
 }
 
+func getDescriptionOfDeniedError(eventWatch watchapi.Interface, ref imageapi.DockerImageReference, err error) error {
+	glog.V(4).Infof("Push has been denied. Waiting for an event with proper error description.")
+	timeout := time.After(time.Second * 2)
+
+waitLoop:
+	for {
+		select {
+		case e := <-eventWatch.ResultChan():
+			if e.Type != watchapi.Added {
+				continue
+			}
+			event, ok := e.Object.(*kapi.Event)
+			if !ok {
+				continue
+			}
+			if event.InvolvedObject.Kind != "ImageStream" || event.InvolvedObject.Namespace != ref.Namespace || event.InvolvedObject.Name != ref.Name {
+				continue
+			}
+
+			glog.V(4).Infof("Replacing error message %q with %q", err.Error(), event.Message)
```

Comment:
How do you know this isn't just the "hey, my image stream is awesome and I tagged unrelated image" event?